pandas/tests/series/indexing/test_where.py

```diff
@@ -452,3 +452,19 @@ def test_where_empty_series_and_empty_cond_having_non_bool_dtypes():
     ser = Series([], dtype=float)
     result = ser.where([])
     tm.assert_series_equal(result, ser)
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_where_nullable_boolean_mask(inplace):
+    # https://github.com/pandas-dev/pandas/issues/35429
+    ser = Series([1, 2, 3])
+    mask = Series([True, False, None], dtype="boolean")
+    expected = Series([1, 999, 999])
+
+    if inplace:
+        result = ser.copy()
+        result.where(mask, 999, inplace=True)
+    else:
+        result = ser.where(mask, 999, inplace=False)
```

Comment:
following on from previous comment.

```
>>> ser = pd.Series([1, 2, 3], dtype="Int64")
>>> mask = pd.Series([True, False, None], dtype="boolean")
>>> ser.where(mask, 999)
0      1
1    999
2    999
dtype: Int64
```
using a nullable mask with a nullable type gives the wrong output?

Reply:
This output is correct IMO, but the output changes if the operation is inplace:
```python
[ins] In [4]: s.where(mask, 999)
Out[4]: 
0      1
1    999
2    999
dtype: Int64

[ins] In [5]: s.where(mask, 999, inplace=True)

[ins] In [6]: s
Out[6]: 
0      1
1    999
2      3
dtype: Int64
```